tools/cleanup.sh

```diff
@@ -1,5 +1,14 @@
-#!/bin/sh
+#!/bin/bash
 #
 # list packages that are not required by packages in this repo
-#
-exec opam admin filter --dry-run --resolve=xs-toolstack --resolve upstream-extra-dummy --resolve depext.transition --resolve opam-ed.0.3 --resolve ocaml-base-compiler.4.08.1 --resolve ppx_tools.6.0+4.08.0 --resolve ocaml-system.4.08.1 --or --with-test
+
+# with-test only looks at test dependencies of the packages listed on the
+# cmdline, not everything recursively
+# so find out recursive deps of xs-toolstack: we want to run tests for all of
+# those (but we don't necessarily need to run the tests of packages pulled in
+# only for testing - that is sometimes not all solvable with same versions)
+set -euo pipefail
+COMPILER=4.14.0
```

Comment:
This version is not yet available, it would be better not to hardcode it. this value should be taken from the CI variable that determines the ocaml version used, or load the env variables from https://github.com/xapi-project/xs-opam/blob/master/tools/xs-opam-ci.env

Reply:
Will address in [the other PR ](https://github.com/xapi-project/xs-opam/pull/627)